akka-stream-typed/src/main/scala/akka/stream/typed/scaladsl/ActorSource.scala

```diff
@@ -59,4 +59,29 @@ object ActorSource {
         bufferSize,
         overflowStrategy)
       .mapMaterializedValue(actorRefAdapter)
+
+  /**
+   * Creates a `Source` that is materialized as an [[akka.actor.ActorRef]].
+   * Messages sent to this actor will be emitted to the stream if there is demand from downstream,
+   * and a new message will only be accepted after the previous messages has been consumed and acknowledged back.
+   * The stream will complete with failure if a message is sent before the acknowledgement has been replied back.
+   *
+   * The stream can be completed with failure by sending a message that is matched by `failureMatcher`. The extracted
+   * [[Throwable]] will be used to fail the stream. In case the Actor is still draining its internal buffer (after having received
+   * a message matched by `completionMatcher`) before signaling completion and it receives a message matched by `failureMatcher`,
+   * the failure will be signaled downstream immediately (instead of the completion signal).
+   *
+   * The actor will be stopped when the stream is completed, failed or canceled from downstream,
+   * i.e. you can watch it to get notified when that happens.
+   */
+  def actorRefWithAck[T](
+      ackMessage: Any,
```

Comment:
shouldn't this be like:
```
def actorRefWithAck[T, Ack](
  ackTo: ActorRef[Ack],
  ackMessage: Ack,
  ...
```

since there is no `sender()` in Typed.

Also meaning that there is a bug in the stage impl. The `ActorRef` in `case e: (ActorRef, T)` will be deadLetters.
Worth adding a test.

Reply:
Right, good to know. I've extended the typed api with ack ref (+spec) and a java api.